lib/BuildSystem/ExternalCommand.cpp

```diff
@@ -223,10 +219,11 @@ void ExternalCommand::start(BuildSystemCommandInterface& bsci,
 
 void ExternalCommand::providePriorValue(BuildSystemCommandInterface&,
                                         core::Task*,
-                                        const BuildValue& value) {
+                                        const BuildValue& value,
+                                        const basic::CommandSignature& signature) {
   if (value.isSuccessfulCommand()) {
     hasPriorResult = true;
-    priorResultCommandSignature = value.getCommandSignature();
```

Comment:
This is kind of ugly. I think that it would be preferable for tasks to not use the signature, since the engine has already decided it should run.   However, the 'can-update-if-newer' bits below rely on this to internally decide to skip the computation anyway.